roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -97,43 +97,37 @@ impl JDsMempool {
 
     pub async fn update_mempool(self_: Arc<Mutex<Self>>) -> Result<(), JdsMempoolError> {
         let mut mempool_ordered: HashMap<Txid, Option<Transaction>> = HashMap::new();
+
         let client = self_
             .safe_lock(|x| x.get_client())
             .map_err(|e| JdsMempoolError::PoisonLock(e.to_string()))?
             .ok_or(JdsMempoolError::NoClient)?;
-        let new_mempool: Result<HashMap<Txid, Option<Transaction>>, JdsMempoolError> = {
-            let self_ = self_.clone();
-            tokio::task::spawn(async move {
-                let mempool: Vec<String> = client
-                    .get_raw_mempool()
-                    .await
-                    .map_err(JdsMempoolError::Rpc)?;
-                for id in &mempool {
-                    let key_id = Txid::from_str(id).unwrap();
-                    let tx = self_.safe_lock(|x| match x.mempool.get(&key_id) {
-                        Some(entry) => entry.clone(),
-                        None => None,
-                    });
-                    let id = Txid::from_str(id).unwrap();
-                    mempool_ordered.insert(id, tx.unwrap());
-                }
-                if mempool_ordered.is_empty() {
-                    Err(JdsMempoolError::EmptyMempool)
-                } else {
-                    Ok(mempool_ordered)
-                }
-            })
-            .await
-            .map_err(JdsMempoolError::TokioJoin)?
-        };
-        match new_mempool {
-            Ok(new_mempool_) => {
-                let _ = self_.safe_lock(|x| {
-                    x.mempool = new_mempool_;
-                });
-                Ok(())
-            }
-            Err(a) => Err(a),
+
+        let mempool: Vec<String> = client.get_raw_mempool().await.map_err(|err| {
+            let err_msg = format!("Error occurred during deserialization: {:?}", err);
+            JdsMempoolError::Rpc(RpcError::Deserialization(err_msg))
+        })?;
+        for id in &mempool {
+            let key_id = Txid::from_str(id).map_err(|err| {
+                let err_msg = format!("Error occurred during deserialization: {:?}", err);
+                JdsMempoolError::Rpc(RpcError::Deserialization(err_msg))
+            })?;
+
+            let tx = self_
+                .safe_lock(|x| match x.mempool.get(&key_id) {
+                    Some(entry) => entry.clone(),
+                    None => None,
+                })
+                .map_err(|e| JdsMempoolError::PoisonLock(e.to_string()))?;
```

Comment:
This is superfluous, as above, when there is a `TryFom ErrorA to ErrorB` as in this case for JdsMempoolError  and PoinsonLock you can just use the question mark to immidatlyb return the error (you are implicity calling into())

Reply:
Are suggesting to add an implementation of try_from that converts a PoisonLock to the relative variant in JdsMempolError? otherwise I cannot see how you can "use the question mark to immidatly"

Reply:
we already have the try from you can just use ?

Reply:
sorry the From not TryFrom

Reply:
a ok is another kind of error you are not returning JdsError but a JdsMempoolError that do not have the from. Yes I suiggest to implement it   in  that case as we have for the JdsError